slmigrate/arghandler.py

```diff
@@ -3,50 +3,43 @@
-    parser.add_argument("--" + constants.states.arg, "--state", help="Migrate system states", action="store_true")
-    parser.add_argument("--" + constants.migration_arg, "--directory", "--folder", help="Specify the directory used for migrated data", action="store", default=constants.migration_dir)
-    parser.add_argument("--" + constants.thdbbug.arg, help="Migrate tag history data to the correct MongoDB to resolve an issue introduced in SystemLink 2020R2 when using a remote Mongo instance. Use --sourcedb to specify a source database. admin is used if none is specfied", action="store_true")
-    parser.add_argument("--" + constants.source_db_arg, "--sourcedb", help="The name of the source directory when performing intra-databse migration", action="store", default=constants.source_db)
-    return parser
-
+def parse_arguments():
+    parser = argparse.ArgumentParser(prog='slmigrate')
+
+    parent_parser = argparse.ArgumentParser(add_help=False)
+    parent_parser.add_argument("--" + constants.tag.arg, "--tags", "--tagingestion", "--taghistory", help="Migrate tags and tag histories", action="store_true", )
+    parent_parser.add_argument("--" + constants.opc.arg, "--opcua", "--opcuaclient", help="Migrate OPCUA sessions and certificates", action="store_true")
+    parent_parser.add_argument("--" + constants.fis.arg, "--file", "--files", help="Migrate ingested files", action="store_true")
+    parent_parser.add_argument("--" + constants.testmonitor.arg, "--test", "--tests", "--testmonitor", help="Migrate Test Monitor data", action="store_true")
+    parent_parser.add_argument("--" + constants.asset.arg, "--assets", help="Migrate asset utilitization and calibration data", action="store_true")
+    parent_parser.add_argument("--" + constants.repository.arg, "--repo", help="Migrate packages and feeds", action="store_true")
+    parent_parser.add_argument("--" + constants.alarmrule.arg, "--alarms", "--alarm", help="Migrate Tag alarm rules", action="store_true")
+    parent_parser.add_argument("--" + constants.userdata.arg, "--ud", help="Migrate user data", action="store_true")
+    parent_parser.add_argument("--" + constants.notification.arg, "--notifications", help="Migrate notifications strategies, templates, and groups", action="store_true")
+    parent_parser.add_argument("--" + constants.states.arg, "--state", help="Migrate system states", action="store_true")
+    parent_parser.add_argument("--" + constants.migration_arg, "--directory", "--folder", help="Specify the directory used for migrated data", action="store", default=constants.migration_dir)
+    parent_parser.add_argument("--" + constants.source_db_arg, "--sourcedb", help="The name of the source directory when performing intra-databse migration", action="store", default=constants.source_db)
+
+    commands = parser.add_subparsers(dest='action')
```

Comment:
Can we store the "action" attribute off as a const? It looks like it is being used as a raw string in multiple locations.

Reply:
I agree with this.